### PR TITLE
Fix name mangling when building with g++ (HASWELL only, WIP)

### PR DESCRIPTION
**Pull request by zanellia**
In order for the `extern C` directive to work correctly, the headers containing it must be included by the file implementing the functions. This PR fixes some issues with name mangling by including the headers containing function declaration in source file containing the corresponding definitions. *Fixes have been tested for HASWELL*, if this PR is approved I can do the same for all targets.

### CODE REVIEW

File: auxiliary/d_aux_lib.c [hunk 1]
```diff
@@ -45,6 +45,8 @@
 #include <math.h>
 
 #include "../include/blasfeo_common.h"
+#include "../../include/blasfeo_d_aux.h"
```
Comment (giaf) [inline auxiliary/d_aux_lib.c]: why 2 folders up?

File: auxiliary/x_aux_lib.c [hunk 2]
```diff
@@ -33,6 +33,8 @@
 *                                                                                                 *
 **************************************************************************************************/
 
+#include "../../include/blasfeo_d_aux.h"
```
Comment (giaf) [inline auxiliary/x_aux_lib.c]: I guess it should be only 1 folder up

**Comment by giaf** [comment]
Once fixed the stuff above, it looks fine, at least by looking at it.

File: auxiliary/d_aux_lib.c (hunk 1 above)
Comment (zanellia) [reply auxiliary/d_aux_lib.c]: good catck, fixed here 
1d003a4

File: auxiliary/x_aux_lib.c (hunk 2 above)
Comment (zanellia) [reply auxiliary/x_aux_lib.c]: fixed here 1d003a4

**Comment by zanellia** [comment]
> Once fixed the stuff above, it looks fine, at least by looking at it.

Sounds good. Would it make sense to do it for all targets (and check that there are no other undersolved symbols for HASWELL too) then?

**Comment by giaf** [comment]
> Would it make sense to do it for all targets (and check that there are no other undersolved symbols for HASWELL too) then?

Yes please, no need to leave half stuff still broken :p

**Pull request merged**